packages/cli/src/run.js

```diff
@@ -33,12 +33,20 @@ export default async function run (_argv) {
     cmd = await getCommand('dev')
   }
 
+  // Check for dev
+  const dev = argv[0] === 'dev'
+
+  // Call setup hook
+  if (typeof hooks.setup === 'function') {
+    await hooks.setup({ cmd, dev, argv })
```

Comment:
If setup hook has been called before hook registration, does it need to be registered again?

Reply:
Indeed this is a special hook which will be called before command initialization and will be never called again We can remove it from object taught if you want :)

Reply:
done